src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import cx from "classnames";
+import React, { FC, ReactElement, useEffect, useState } from "react";
+import { SystemTheme, Theme, ThemeColor } from "./ThemeProviderConstants";
+import { generateRandomAlphaString, generateThemeCssOverride, shouldGenerateTheme } from "./ThemeProviderUtils";
+import { withStaticProps } from "../../types";
+
+export interface ThemeProviderProps {
+  /**
+   * The theme to apply, consists of name - name of css class that will be added to the children, which should be unique - and the object of colors overrides for each system theme.
+   */
+  theme?: Theme;
+  /**
+   * The children to render with the theme
+   */
+  children: ReactElement;
+  /**
+   * Random string which adds up to theme name to make it more specific (in case if theme.name is colliding with some other class name)
+   */
+  randomStringSelector?: string;
+}
+
+const ThemeProvider: FC<ThemeProviderProps> & {
+  systemThemes?: typeof SystemTheme;
+  colors?: typeof ThemeColor;
+} = ({ theme, children, randomStringSelector: customRandomStringSelector }) => {
+  const [stylesLoaded, setStylesLoaded] = useState(false);
+  // Using random string selector to secure selector by making it more specific - if theme name is not unique
+  const randomStringSelector = useState(customRandomStringSelector || generateRandomAlphaString())[0];
```

Comment:
@orrgottlieb wanted to add this to be more secure in cases where e.g. when theme.name is the same as some other class name - so it would still work
Let's rename it to `additionalStringSelector`? wdyt @shlomitc ?

Reply:
Done